File: example-java/src/main/java/com/github/simonpercic/oklogexamplejava/DataFetcher.java
package com.github.simonpercic.oklogexamplejava;

import com.github.simonpercic.oklogexamplejava.data.api.model.response.show.ShowResponse;
import com.github.simonpercic.oklogexamplejava.data.api.okhttp.DataModule;
import com.github.simonpercic.oklogexamplejava.data.api.okhttp.RestApi;

import java.util.List;

import javax.inject.Inject;

/**
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
class DataFetcher {

    @Inject RestApi restApi;

    void fetch() {
        DaggerDataFetcherComponent.builder()
                .dataModule(new DataModule())
                .build()
                .inject(this);

        List<ShowResponse> response = restApi.getShows();

        System.out.printf("Got %d shows%n", response.size());
    }
}

File: example-kotlin/src/main/kotlin/com/github/simonpercic/oklogexamplekotlin/DataFetcher.kt
package com.github.simonpercic.oklogexamplekotlin

import com.github.simonpercic.oklogexamplekotlin.data.api.okhttp.DataModule
import com.github.simonpercic.oklogexamplekotlin.data.api.okhttp.RestApi
import javax.inject.Inject

/**
 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
class DataFetcher {

    @Inject
    lateinit var restApi: RestApi

    fun fetch() {
        DaggerDataFetcherComponent.builder()
            .dataModule(DataModule())
            .build()
            .inject(this)

        val response = restApi.getShows()

        System.out.printf("Got %d shows%n", response.size)
    }
}
